src/headerssync.cpp

```diff
@@ -193,7 +186,7 @@ bool HeadersSyncState::ValidateAndProcessSingleHeader(const CBlockHeader& curren
         return false;
     }
 
-    if (next_height % HEADER_COMMITMENT_PERIOD == m_commit_offset) {
+    if (next_height % m_params.commitment_period == m_commit_offset) {
```

Comment:
a4191b57b21a0fea871206f368972c7741ad6437 is a bit complicated, can you split it up to smaller refactors?